src/audio/component.c

```diff
@@ -62,8 +62,8 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 	int requested_state = comp_get_requested_state(cmd);
```

Comment:
performance difference is very much test-dependent. If you do a lot of starting / stopping with very short streaming times, or if you do parallel asynchronous multi-PCM testing, your logging contribution will be rather high, however if you just start a single stream, let it run for a minute, then stop - the saving will be much lower, because obviously prints are done during set up and tear down and very few during streaming

Reply:
That's correct, and I also agree that some logs are useful, just one thinking: how about use debug build to perform test(with dbg level log print), and keep release build as much less logs?  I guess there maybe something wrong for timing due to too much dbg level logs, it may cause timeout for some cases.

Bios does not have any logs for production build, and validation team use debug build perform daily test.

Reply:
unfortunately we have too mane debug prints, so if you enable them all, debugging becomes way too difficult. I think it enables printing during streaming, so your log is flooded, messages get dropped, or the whole run slows down to timeouts, etc. Maybe we need verbose debug for debugging during streaming to distinguish it from "slightly more verbose than usual but still usable debug logging" but we don't have it so far

Reply:
@btian1 Yup, @lyakh has a point. In practical terms, if you want to get these logs back, that is not easy as current "dbg" is so crowded. But I do think we have too much at "info" so this PR is needed. It's now a question of going through the log entries and evaluating per log, whether it should be kept or not.

@btian1 And one can of course ship with traces disabled, that's just a matter of a Kconfig option. Some SOF release binaries do not have any logs enabled. But e.g. the public upstream SOF releases for Intel hardware do have logs enabled as the performance impact has been within acceptable limits. But at code level, we should not assume anything about this, but rather enable both options to be used.

Reply:
ok, let's work step by step to get logging better.